test/test_dependency_versions.py

```diff
@@ -58,6 +59,8 @@ def test_pinned_versions(tmp_path, python_version, build_frontend_env):
     build_environment = {}
 
     if python_version == "3.6":
+        if utils.platform == "macos" and platform.machine() == "arm64":
+            pytest.skip("macOS arm64 does not support Python 3.6")
```

Comment:
This is confusing me... 3.6 should run under Rosetta I think? It does on my M1:

```
$ which python3.6
/Library/Frameworks/Python.framework/Versions/3.6/bin/python3.6
$ python3.6 --version
Python 3.6.8
$ python3.6 -m platform
Darwin-21.5.0-x86_64-i386-64bit
```

Reply:
Sure but x86_64 target is not enabled by default when running on arm64. I suppose I should try enabling that, at least for the test suite.

Reply:
I thought about this a bit more and I think that the default (`auto`) architectures built on Apple Silicon should include `x86_64` since Rosetta is a thing and current default is `arm64` + `universal2` which is not ideal for older Pythons (which simply don't support Apple Silicon natively) as well as older pip versions.

Is there any benefit to building `x86_64` + `universal2` wheels over building `x86_64` + `arm64`? I feel like a reasonable default (`auto`) would be to build `x86_64` + `arm64`.
However, I suppose `x86_64` + `universal2` is fine too, just makes it a kind of an untypical case where we wouldn't be building wheels for the native architecture as we do in all the other cases. There's also an option of building `x86_64` + `universal2` + `arm64` but cibuildwheel's documentation seems to somewhat discourage that (+ it would increase build time somewhat, though building wheels on Apple Silicon takes the least time anyway :smile:).

Reply:
So I've tried changing `auto` to include all of the arches - `x86_64`, `arm64`, and `universal2`. It required some changes to the generated wheel file names in the `expected_wheels()` utility in the test suite and temporarily skipping (only for the sake of testing this) two tests that don't work properly because of #1193.
Once all of that was done, it worked flawlessly without the need for *this* adjustment in the test suite as well as another one that I had to make in this PR.
So if the decision on which architectures `auto` should build on macOS arm64 would be made, it shouldn't be a big issue to adjust this PR appropriately (presumably no changes other than reverts of unnecessary test suite changes would be required if the change to `auto` was done in a separate PR).

Diff: https://github.com/jack1142/cibuildwheel/compare/add_cirrus_ci_support...jack1142:cibuildwheel:try_x86
Build log: https://cirrus-ci.com/task/5300735129681920?logs=run_cibuildwheel_tests

Reply:
Sorry, I forgot that the default `archs` would mean that Python 3.6/3.7 build nothing on arm64. 

The question of the default is a tricky one. My view when this decision was made is that we're designing defaults for a world where arm64 and x86_64 runners coexist. The best situation is to build+test x86_64 natively and arm64 natively, rather than using emulation - if we can help it. 

I'm still assuming that we're going to get arm64 runners on the major CI providers at some point. At that point, I think that the recommended configuration will have both a x86_64 runner and a arm64 runner. So I think we should design defaults that 'just work' for that.

That's where the current defaults come from. Since CirrusCI only has M1 runners, I can see that we might include an option in our example config: `CIBW_ARCHS: x86_64, arm64, universal2`.

The question of whether to default building both `universal2` and `arm64` on the arm runner is still open, in my mind. I'm still mostly of the opinion that building both is good, because universal2 packages are preferred in some contexts (e.g. distributable GUI apps), and that most packages are small and the extra build time isn't much of a hit. Of course, people building big libraries would be wise to change this to just build arm64.

Reply:
I imagine we would want the test suite to be altered so that on macOS arm64 it runs x86_64 builds in majority of the tests as well to allow detection of issues such as #1193. I can probably look into that.

Reply:
Okay, so what do you think should be done about this?

Reply:
Personally, I think the default should be native only. Eventually no one will care about Intel / universal and will only produce native binaries again. Also, there are still workarounds related to cross compiling for universal (like requiring all dependencies to also be universal). And we don’t support it yet, but there are two ways to make a universal binary, the default way and merging two native binaries. And finally universal vs. native is a choice, and I’d you pick universal, you don’t need native.

So I’d recommend native being default, and only producing native binaries with it.

Reply:
> Also, there are still workarounds related to cross compiling for universal (like requiring all dependencies to also be universal)

Certainly the idea that we're forcing _more_ work on packagers to support universal2 cross-compilation alongside the native arm64 seems like a strong argument against universal2 by-default.

Yes, that seems like a well-reasoned position @henryiii. I'd be happy to go down the native-only route, too.

Reply:
> Okay, so what do you think should be done about this?

Even if we do end up changing the default archs on arm64, this PR isn't the place to do it. I've created #1204 for that. Lets keep the archs defaults the same here.

Reply:
We ended up with only building arm64 wheels by default (#1204) so this skip needs to remain here as Python 3.6 can only run on macOS arm64 through Rosetta which uses x86_64 wheels.

Reply:
Are you sure this needs to be here? The exact same situation occurs for Python 3.7, there is no (official) AS support, but a skip wasn’t added there.

Reply:
It's a confusingly written test - `3.7` is not part of `pytest.mark.parametrize` and so it isn't actually being run while `3.6` is.

This is the only part of the test that mentions 3.7, it's not mentioned in the `if` at the end of the test nor in `parametrize`. Do you want me to add or remove 3.7 from the test?

Reply:
Just to add onto this, here's how it fails on 3.6:
https://cirrus-ci.com/task/5307329800306688?logs=run_cibuildwheel_tests#L868